README_SELFHOSTED_RUNNERS.md

```diff
@@ -0,0 +1,192 @@
+ # Self-hosted runners
+
+ A runner is a server that runs your workflows when they're triggered. Each runner can run a single job at a time. GitHub provides Ubuntu Linux, Microsoft Windows, and macOS runners to run your workflows; each workflow run executes in a fresh, newly-provisioned virtual machine. If you need a different operating system or require a specific hardware configuration, you can host your [own runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners).
+ A self-hosted runner is a system that you deploy and manage to execute jobs from GitHub Actions on GitHub.com. 
+
+## Prerequisites:
+
+  ### 1. Create VMs to use as self-hosted runners
+
+  - To create VMs in `Proxmox` use [scripts](https://github.com/Alliedium/awesome-proxmox/tree/main/vm-cloud-init-shell#vm-provisioning-scripts-based-on-cloud-init-images)
+
+  - To create VMs in `Hyper-V` use [steps](https://github.com/Alliedium/awesome-devops/tree/main/03_virtualization_on_windows_and_zfs_11-aug-2022#create-vms-in-hyper-v)
+
+  ### 2. Install `docker` if you want to run workflow jobs in containers
+
+  -  [`Manjaro`](https://github.com/Alliedium/awesome-linux-config/blob/master/manjaro/basic/install_docker.sh)
+
+  -  [Ubuntu](https://docs.docker.com/engine/install/ubuntu/)
+
+
+  ### 3. Install [tmate](https://tmate.io/) for debugging
+
+ 
+## Repository-level runners
+
+### 1. Create a fork from the [main repository](https://github.com/Alliedium/awesome-github-actions)
+
+![fork](./images/fork.png)
+
+### 2. Clone [project](https://github.com/Alliedium/awesome-github-actions)
+
+```
+git clone https://github.com/Alliedium/awesome-github-actions.git $HOME/awesome-github-actions
+```
+
+Replace the `https://github.com/Alliedium/awesome-github-actions.git`  repository URL with the fork URL you created
+
+### 3. Add a self-hosted Linux runner to the `GitHub` repository.
+
+![add_runner](./images/add_runner.png)
+
+Run commands on the Linux VM `runner` 
+
+- Create a `actions-runner` folder
+
+```
+mkdir actions-runner && cd actions-runner
+```
+
+- Download the latest runner package
+
```

Comment:
To find the latest version of packages, use [lastversion](https://github.com/dvershinin/lastversion) tool.

Run command 
`lastversion --assets https://github.com/actions/runner/releases/download --filter "actions-runner-linux-x64-(\d{0,3}\.\d{0,3}\.\d{0,3}).tar.gz"`

You get the URL of the `actions-runner-linux-x64` resource of the latest version for downloading the file.
In our case `2.305.0` the latest vesion.